src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineView.xaml.cs

```diff
@@ -0,0 +1,196 @@
+            if (wpfTextView is null)
+                return VSConstants.E_FAIL;
+
+            _trackedTextViews.Add(textView, wpfTextView);
+
+            // In the split window case, there's two views (each with its own caret position) but only one text buffer.
+            // Subscribe to caret position changes once per view.
+            wpfTextView.Caret.PositionChanged += Caret_PositionChanged;
+
+            return VSConstants.S_OK;
+        }
+
+        private void ExpandAll(object sender, RoutedEventArgs e)
+            => _viewModel.EnqueueExpandOrCollapse(ExpansionOption.Expand);
+
+        private void CollapseAll(object sender, RoutedEventArgs e)
+            => _viewModel.EnqueueExpandOrCollapse(ExpansionOption.Collapse);
+
+        private void SearchBox_TextChanged(object sender, EventArgs e)
+        {
+            var newText = SearchBox.Text ?? string.Empty;
+            _viewModel.EnqueueFilter(newText);
+        }
+
+        private void SortByName(object sender, EventArgs e)
+            => UpdateSort(SortOption.Name);
+
+        private void SortByOrder(object sender, EventArgs e)
+            => UpdateSort(SortOption.Location);
+
+        private void SortByType(object sender, EventArgs e)
+            => UpdateSort(SortOption.Type);
+
+        private void UpdateSort(SortOption sortOption)
+        {
+            // Log which sort option was used
+            Logger.Log(sortOption switch
+            {
+                SortOption.Name => FunctionId.DocumentOutline_SortByName,
+                SortOption.Location => FunctionId.DocumentOutline_SortByOrder,
+                SortOption.Type => FunctionId.DocumentOutline_SortByType,
+                _ => throw new NotImplementedException(),
+            }, logLevel: LogLevel.Information);
+
+            // Defer changes until all the properties have been set
+            using (var _1 = _collectionView.DeferRefresh())
+            {
+                // Update top-level sorting options for our tree view
+                _collectionView.SortDescriptions.UpdateSortDescription(sortOption);
+
+                // Set the sort option property to begin live-sorting
+                _viewModel.SortOption = sortOption;
+            }
+
+            // Queue a refresh now that everything is set.
+            _collectionView.Refresh();
+        }
+
+        /// <summary>
+        /// When a symbol node in the window is clicked, move the caret to its position in the latest active text view.
+        /// </summary>
+        private void SymbolTree_MouseDown(object sender, EventArgs e)
+        {
+            if (sender is StackPanel panel && panel.DataContext is DocumentSymbolDataViewModel symbol)
+            {
+                _isNavigating = true;
+                try
+                {
+                    _codeWindow.GetLastActiveView(out var textView);
+                    Assumes.NotNull(textView);
```

Comment:
worrying.  are we 100% sure this can't happen?  if so, doc why we feel this certain.

Reply:
The contract with the editor is that we are disposed once the active code editor is closed or focus changed to a different tab/window. For this to be null would mean that the developer has closed a window but we have not been disposed and the developer then selects elements within us. I will add a comment explaining this.

Reply:
@jasonmalinowski you mentioned a way to extract out the logic to be able to give us the right snapshots and whatnot.